library/scripts/jjb_compare_xml.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+set -ex
+
+# workaround for old RHELs, we need to install correct tox system-wide
+rm -rf venv
+virtualenv venv
+source venv/bin/activate
+pip install pip --upgrade
+pip install tox
+# ---
+
+JOBS_OUT_DIR=${WORKSPACE}/output/jobs
+JOBS_LOGFILE=${JOBS_OUT_DIR}/jobs-diff.log
+RESULT=''
+BLOCKLIST=blocklist
+
+# First generate output from BASE_COMMIT vars value
+git checkout "${targetBranch}"
+
+rm -rf .tox
+for ENV in ${WORKSPACE}/servers/*; do
+  tox -e "${ENV##*/}"
+  mkdir -p "./output/jobs/old"
+  mv "./output/${ENV##*/}" "./output/jobs/old/${ENV##*/}"
+done
+
+# Then use that as a reference to compare against HEAD
+git checkout "origin/pr/${pullRequestId}/merge"
+rm -rf .tox
+for ENV in ${WORKSPACE}/servers/*; do
+  tox -e "${ENV##*/}"
+  mkdir -p "./output/jobs/new"
+  mv "./output/${ENV##*/}" "./output/jobs/new/${ENV##*/}"
+done
+
+compare_xml() {
+
+# Replace arguments with built-in variables ($1 - path to jobs or views output directory, $2 - path to jobs or views log file)
+OUT_DIR=$1
+LOGFILE=$2
+# Specifying for http links type of comaprison (jobs or views)
+TYPE=$3
+
+BLOCK=0
+CHANGE=0
+ADD=0
+REMOVE=0
+
+BLOCKED="[blocked]<br>"
+CHANGED="[changed]<br>"
+ADDED="[added]<br>"
+REMOVED="[removed]<br>"
+
+DIFF=$(diff -q -r -u "${OUT_DIR}/old" "${OUT_DIR}/new" &>"${LOGFILE}"; echo "${?}")
```

Comment:
The layout of output folder is out of scope of the compare+xml function. Thus instead of  `${OUT_DIR}`, `${OUT_DIR}/old` and `${OUT_DIR}/old` we should have `FROM_DIR` and `$TO_DIR` or A_DIR and B_DIR as diff usually uses this naming.

I would also move comparison function to a separate utils/compare.sh script. And call it in the JJB step as ./utils/compare.sh output/new output/old > report.html